### PR TITLE
Part 2

### PR DESCRIPTION
**Pull request by benjimarshall**
Extended FizzBuzz game, with Bangs, Bongs, Fezzs and reversing on 17s.

Built around a list of strings for nicer "Fezz" insertion and easier reversing of the list under modulo 17

### CODE REVIEW

File: FizzBuzz/Program.cs [hunk 1]
```diff
@@ -9,23 +9,50 @@ static void Main(string[] args)
         {
             for (int i = 1; i <= 100; i++)
             {
-                StringBuilder result = new StringBuilder();
+                List<string> results = new List<string>();
```
Comment (FridaTveit) [inline FizzBuzz/Program.cs:12]: Could you just use `var` here instead of `List<string>`? 🙂

File: FizzBuzz/Program.cs [hunk 2]
```diff
@@ -9,23 +9,50 @@ static void Main(string[] args)
...
+
+                if (i % 11 == 0)
+                {
+                    results.Clear();
+                    results.Add("Bong");
+                }
+
+                if (i % 13 == 0)
+                {
+                    var index = results.FindIndex(s => s.StartsWith("B"));
```
Comment (FridaTveit) [inline FizzBuzz/Program.cs:36]: Having both `i` and `index` is a bit confusing. Can you think of a more descriptive name for `index`? 🙂 

Or you could also extract the logic to find the index to a separate method. I might do that as this method is getting rather long. But it's up to you, not a major thing 🙂

File: FizzBuzz/Program.cs [hunk 3]
```diff
@@ -9,23 +9,50 @@ static void Main(string[] args)
...
+
+                    results.Insert(index, "Fezz");
+                }
+
+                if (i % 17 == 0)
+                {
+                    results.Reverse();
+                }
+
+                if (results.Count == 0)
```
Comment (FridaTveit) [inline FizzBuzz/Program.cs:50]: Could use `.Any()` to check if the list is empty 🙂

**Review by FridaTveit — COMMENTED** [review]
A few minor comments otherwise looks great! 🙂

File: FizzBuzz/Program.cs (hunk 1 above)
Comment (benjimarshall) [reply FizzBuzz/Program.cs]: Yep, fixed in f0493ec :)

File: FizzBuzz/Program.cs (hunk 2 above)
Comment (benjimarshall) [reply FizzBuzz/Program.cs]: Yes, this is something I was beginning to do for the extension tasks and the main method was definitely ready to be broken up. Fixed in 05e32c9

File: FizzBuzz/Program.cs [hunk 4]
```diff
@@ -9,51 +9,56 @@ static void Main(string[] args)
...
-                {
-                    results.Add(i.ToString());
-                }
-
-                Console.WriteLine(String.Join("", results));
+                Console.WriteLine(fizzbuzz(i));
             }
         }
+
+        static string fizzbuzz(int i)
```
Comment (FridaTveit) [inline FizzBuzz/Program.cs:17]: In C# it's best practice to start method names with a capital letter 🙂

**Review by FridaTveit — COMMENTED** [review]
One tiny thing otherwise looks great. Nice refactoring 🙂

**Review by FridaTveit — APPROVED** [review]
Looks great 🙂

File: FizzBuzz/Program.cs (hunk 4 above)
Comment (benjimarshall) [reply FizzBuzz/Program.cs]: Yes, sorry my Java habits are showing through! Fixed in 798648b

**Pull request merged**